oa-enterprise/spec/omniauth/strategies/cas_spec.rb

```diff
@@ -45,42 +45,79 @@ def strategy
   end
 
   describe 'GET /auth/cas/callback with a valid ticket' do
-    before do
-      stub_request(:get, /^https:\/\/cas.example.org(:443)?\/serviceValidate\?([^&]+&)?ticket=593af/).
-         with { |request| @request_uri = request.uri.to_s }.
-         to_return(:body => File.read(File.join(File.dirname(__FILE__), '..', '..', 'fixtures', 'cas_success.xml')))
-      get '/auth/cas/callback?ticket=593af'
-    end
+    shared_examples :successful_validation do
+      before do
+        stub_request(:get, /^http(s)?:\/\/cas.example.org(:8080)?\/serviceValidate\?([^&]+&)?ticket=593af/)
+          .with { |request| @request_uri = request.uri.to_s }
+          .to_return( body: File.read("spec/fixtures/#{xml_file_name}") )
 
-    it 'should strip the ticket parameter from the callback URL before sending it to the CAS server' do
-      @request_uri.scan('ticket=').length.should == 1
-    end
+          get "/auth/cas/callback?ticket=593af&url=#{return_url}"
+      end
+
+      it 'should strip the ticket parameter from the callback URL before sending it to the CAS server' do
```

Comment:
most of these tests are from upstream, but some of my own intepretation based on our sample success criteria